### PR TITLE
Clean up PodSpec decoration to honor new validation

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Clean up PodSpec decoration to honor new validation
    
With stricter rules on what can be passed as a PodSpec in a ProwJob, the
decoration functions can be pruned down to a more minimal set of logic.
    
Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/area prow
/cc @cjwagner @kargakis 
/assign @BenTheElder 

The tests may become more problematic as the PodSpec decoration gets more complex...

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
Depends on #7348

File: prow/config/config.go [hunk 1]
```diff
@@ -507,6 +517,30 @@ func validatePresets(name string, labels map[string]string, spec *v1.PodSpec, pr
 	return nil
 }
 
+func validatePodSpec(name string, jobType kube.ProwJobType, spec *v1.PodSpec) error {
+	if spec == nil {
+		return nil
+	}
+
+	if len(spec.InitContainers) != 0 {
```
Comment (fejta) [inline prow/config/config.go]: Can we put a TODO to relax these restrictions later?

File: prow/pod-utils/decorate/podspec.go [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+// to their values that should be available for a job spec
+func EnvForSpec(spec JobSpec) (map[string]string, error) {
+	env := map[string]string{
+		JobNameEnv:   spec.Job,
+		BuildIdEnv:   spec.BuildId,
+		ProwJobIdEnv: spec.ProwJobId,
+		JobTypeEnv:   string(spec.Type),
+	}
+
+	// for backwards compatibility, we provide the build ID
```
Comment (fejta) [inline prow/pod-utils/decorate/podspec.go]: Can we put a plan in place to retire both of these?

Ideally we only have one thing, and ideally we don't have agent specific code in here.

File: prow/config/config.go (hunk 1 above)
Comment (stevekuznetsov) [reply prow/config/config.go]: That would require a radical redesign of what a test is and how `plank` interacts with them, so unless a `//TODO` to do architecture+design makes sense I'd leave it as-is?

**Comment by stevekuznetsov** [comment]
Going to approach this in a slightly different way.

/close

**Pull request closed without merging**